src/main/java/com/ably/kafka/connect/ChannelSinkMapping.java

```diff
@@ -0,0 +1,19 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.realtime.AblyRealtime;
+import io.ably.lib.realtime.Channel;
+import io.ably.lib.types.AblyException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import javax.annotation.Nonnull;
+
+public interface ChannelSinkMapping {
+    /**
+     * Returns the channel for the given sink record.
+     *
+     * @param sinkRecord The sink record.
+     * @param ablyRealtime AblyRealtime instance.
+     * @return The channel.
+     */
+    Channel getChannel(@Nonnull SinkRecord sinkRecord,@Nonnull AblyRealtime ablyRealtime) throws AblyException;
```

Comment:
> This will mean custom implementations can forward a single message to many Ably channels, if we want to support that use case

As far as I am aware we don't have a case where we do a map from single record to multiple channels. If there is such case and we want to return a set, I can change the interface method by providing a channel wrapper and a forwarding publish method that will implement equals / hashcode. 

Also from the memory performance perspective, channels are already maintained on Java SDK and they may not be automatically released. As currently we have a single channel, I didn't want to complicate the situation on channel config class. When we have the case where we could be keeping multiple channels at the same time, we will probably need to keep some maximum number of channels and release others (we can use a LRU cache for that)